src/Codegen/Builders/ObjectBuilder.hack

```diff
@@ -18,7 +18,61 @@ final class ObjectBuilder<TField as IFieldBuilder> extends CompositeBuilder<TFie
         return new ObjectBuilder(
             $type_info,
             $type_alias->getName(),
-            Vec\map_with_key($ts['fields'], ($name, $ts) ==> new ShapeFieldBuilder($name, $ts)),
+            Vec\map_with_key(
+                $ts['fields'],
+                ($name, $ts) ==> FieldBuilder::fromShapeField($name, $type_alias->getName(), $ts),
+            ),
+        );
+    }
+
+    public static function forConnection(string $name): ObjectBuilder {
+        return new ObjectBuilder(
+            new \Slack\GraphQL\ObjectType($name, $name), // TODO: Description
+            $name,  // hack type
+            vec[ // fields
+                new MethodFieldBuilder(shape(
+                    'name' => 'edges',
+                    'method_name' => 'getEdges',
+                    'output_type' => shape(
+                        'type' => 'UserEdge::nonNullable()->nullableOutputListOf()',
+                        'needs_await' => true,
+                    ),
+                    'declaring_type' => $name,
+                    'parameters' => vec[],
+                )),
+                new MethodFieldBuilder(shape(
+                    'name' => 'pageInfo',
+                    'method_name' => 'getPageInfo',
+                    'output_type' => shape('type' => 'PageInfo::nullableOutput()', 'needs_await' => true),
+                    'declaring_type' => $name,
+                    'parameters' => vec[],
+                )),
+            ],
+        );
+    }
+
+    // TODO: It should be possible to create user-defined edges which contain additional fields.
```

Comment:
I was thinking about this, you can always just create the connections and edges yourself if you want this kind of customizability.

Reply:
Adding extra fields to the connection type and/or the edge type is probably worth supporting at some point (especially edge fields, e.g. the timestamp of when the edge was created is very common for all kinds of edges), but it can wait.